### ISSUE TITLE
I want to build and replace the r8125 driver separately

### ISSUE THREAD

**Issue by zelesdev** [friendlyarm/Actions-FriendlyWrt#83]
Hello, I want to build and replace the r8125 driver or r8152 driver on the nanopi r5c board
To enable the giga-lite feature (This is a feature that is required in a special internet environment. I bought an r5c with realtek for that purpose)
In Linux Mint, the build and replacement went smoothly

However, in friendlywrt, the kernel header is missing, so the driver build is impossible, and in friendlycore, the driver build is successful, but the replacement with insmod is not possible even if the vermagic is the same

I also tried building the ipk through the sdk downloaded from gdrive, but the current friendlywrt 23.05 has a kernel version of 6.1.57, and even if I use the latest sdk on the gdrive, the kmod file is built for the kernel version of 5.15.134 and cannot be used

Is there any way? Any hints would be appreciated. I've been struggling for several days, and now I'm stuck.

**Comment by lawrencetg** [comment]
[0001-add-sample-script-to-build-3rd-r8125-driver.patch](https://github.com/user-attachments/files/17744621/0001-add-sample-script-to-build-3rd-r8125-driver.patch)

you may refer to this patch to integrate the r8125 driver into this project.

**Comment by zelesdev** [comment]
Thanks
However, I forked this repo and built a new wrt by referencing and only changing the driver to my repo.
If I do it the way you suggested, is it possible to replace just the driver?

**Comment by lawrencetg** [comment]
> Thanks However, I forked this repo and built a new wrt by referencing and only changing the driver to my repo. If I do it the way you suggested, is it possible to replace just the driver?

yes, you can modify the repo url of the driver module in the script.
